click/_bashcomplete.py

```diff
@@ -39,12 +39,21 @@ def resolve_ctx(cli, prog_name, args):
     return ctx
 
 
-def get_choices(cli, prog_name, args, incomplete):
+def get_choices(cli, prog_name, args, incomplete, last_complete=None):
     ctx = resolve_ctx(cli, prog_name, args)
     if ctx is None:
         return
 
     choices = []
+
+    if last_complete:
+        for param in ctx.command.params:
+            if last_complete in param.opts:
+                if param.completer is not None:
+                    choices.extend(str(c) for c in param.completer())
+                if param.choices is not None:
```

Comment:
I think this extra logic for choices could be moved somewhere else by using the custom completer API you just introduced.